src/object/object_manager.cpp

```diff
@@ -162,16 +164,19 @@ CObject* CObjectManager::CreateObject(Math::Vector pos, float angle, ObjectType
     params.pos = pos;
     params.angle = angle;
     params.type = type;
-    
+    params.power = ClampPower(type,power);
+    return CreateObject(params);
```

Comment:
Here's where the call to the "real" `CreateObject()` function happens where the clamping should actually happen. This method here is only a convenience thing so you don't have to create the ObjectCreateParams yourself. There shouldn't be any logic happening here beyond wrapping the arguments in an ObjectCreateParams object.

Reply:
Ah, I see it clearly now "return CreateObject(params);". Should have paid more attention to that last line. Thanks!

Reply:
Actually, with this removed, bots using produce() always produce power cells with zero energy, so I had to add it back.

Reply:
Was that really the only way to fix this? Did you try debugging why produce() did that?

Reply:
When power cells are created from scene files in robotmain.cpp, it uses CreateObject(params) and the power is clamped by the ClampPower method. That was working. 

In scriptfunc.cpp, produce() uses CreateObject(pos, angle, type, power). You pointed out that CreateObject(params) within this method, so it seems like the value should be clamped just like it is with scene files, but the cells are always produced with zero power.

Hmmmm in the middle of typing this I think I figured it out. Missing params.power = power in the CreateObject(pos, angle, type, power) method. I never put this back after deleting the call to ClampPower.

Pushing fix now. Sorry about that!